hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -134,14 +134,14 @@ class ColumnStats {
 
     HashMap<String, ColumnStats> allColumnStats = new HashMap<>();
 
-    // Collect stats for all columns by iterating through records while accounting
-    // corresponding stats
-    records.forEach((record) -> {
-      // For each column (field) we have to index update corresponding column stats
-      // with the values from this record
-      targetFields.forEach(field -> {
-        ColumnStats colStats = allColumnStats.computeIfAbsent(field.name(), (ignored) -> new ColumnStats());
-
+    // For each column (field) we have to index update corresponding column stats
+    // with the values from this record
+    targetFields.forEach(field -> {
```

Comment:
Why is this changing?

Reply:
Because usually fields number are less than records.

Reply:
We don MxN traversals (and similar amount of lookups) anyway, i don't think this re-ordering changes anything

Reply:
@danny0405 shall we keep the code as before?  Doesn't look like the logic changes anyway.

Reply:
> We don MxN traversals (and similar amount of lookups) anyway, i don't think this re-ordering changes anything

It reduces the invocation of 
```java
final ColumnStats colStats = allColumnStats.computeIfAbsent(field.name(), (ignored) -> new ColumnStats());
```

Reply:
> @danny0405 shall we keep the code as before? Doesn't look like the logic changes anyway.

No, there is no need to generate the `ColumnStats` per-record iteration.

Reply:
@danny0405 it does not generate new `ColumnStats` per record -- `ColumnStats` is generated ones per record. Re-ordering does not change anything

Reply:
I took another look actually and this change doesn't make sense to me: previously we'd iterate over list of records **once**, iterating over fields for every record providing for spatial locality (every record will be accessed just once, all of the fields will be handled w/in the inner loop) which provides for greater CPU caching opportunity.

Now with your change we will make N iterations over all of the records (where N is the number of columns) providing for no spatial locality (we will first access field A from every record, then field B, then field C, etc)

Reply:
Makes sense and rolled it back.